### PR TITLE
Add check for when a task already had a assignee

### PR DESCRIPTION
**Pull request by alexbaeza**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-1047

### Change description ###

Add check for when a task already had a assignee
Validate if user roles are still valid

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by dharmendrak — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostInitiateByIdControllerTest.java [hunk 1]
```diff
@@ -487,61 +481,322 @@ void should_return_201_with_task_assigned() throws Exception {
...
+                    .value(anyOf(is("tribunal-caseworker"), is("senior-tribunal-caseworker"))),
+                jsonPath("$.task_role_resources.[1].read").value(true),
+                jsonPath("$.task_role_resources.[1].own").value(true),
+                jsonPath("$.task_role_resources.[1].execute").value(false),
+                jsonPath("$.task_role_resources.[1].manage").value(true),
+                jsonPath("$.task_role_resources.[1].cancel").value(true),
+                jsonPath("$.task_role_resources.[1].refer").value(true),
+                jsonPath("$.task_role_resources.[1].auto_assignable").value(false)
+            );
     }
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostInitiateByIdControllerTest.java:800]: Can we include a test for AC2, where the Assignee no longer has permission to complete task + autoAssign has no user with permission corresponding to task, then task state should be unAssigned in both CFT and Camunda

File: src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostInitiateByIdControllerTest.java [hunk 2]
```diff
@@ -487,61 +481,322 @@ void should_return_201_with_task_assigned() throws Exception {
...
+                jsonPath("$.task_role_resources.[1].role_name")
+                    .value(anyOf(is("tribunal-caseworker"), is("senior-tribunal-caseworker"))),
+                jsonPath("$.task_role_resources.[1].read").value(true),
+                jsonPath("$.task_role_resources.[1].own").value(true),
+                jsonPath("$.task_role_resources.[1].execute").value(false),
+                jsonPath("$.task_role_resources.[1].manage").value(true),
+                jsonPath("$.task_role_resources.[1].cancel").value(true),
+                jsonPath("$.task_role_resources.[1].refer").value(true),
+                jsonPath("$.task_role_resources.[1].auto_assignable").value(false)
+            );
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostInitiateByIdControllerTest.java:799]: Would including a check that the camunda State for task has been updated with the new Assignee fulfil the AC in relation to this test?

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoAssignmentService.java [hunk 3]
```diff
@@ -115,6 +95,44 @@ public TaskResource autoAssignCFTTask(TaskResource taskResource) {
...
+                taskRoleResource -> taskRoleResource,
+                (a, b) -> b
+            ));
+
+        List<RoleAssignment> orderedRoleAssignments = orderRoleAssignments(rolesList, roleAssignments);
+        return orderedRoleAssignments.stream()
+            .filter(roleAssignment -> {
+                TaskRoleResource taskRoleResource = roleResourceMap.get(roleAssignment.getRoleName());
+                if (taskRoleResource == null
+                    || taskRoleResource.getAuthorizations() == null
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoAssignmentService.java:124]: can we add coverage for this -> taskRoleResource == null
                    || taskRoleResource.getAuthorizations() == null

Comment (alexbaeza) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/taskconfiguration/services/TaskAutoAssignmentService.java:124]: added

**Pull request merged**